package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -59,11 +110,99 @@ def to_json(self) -> str:
                 session_dict[key] = str(value)
         return json.dumps(session_dict)
 
+    def on_commit_sync(self, _):
+        self.sync()
+
     def save(self):
         """Save the session store info on db ."""
         engine, session_class = create_db_engine(self.location)
         Base.metadata.create_all(bind=engine)
         database = next(get_db(session_class))
+        event.listen(database, "after_commit", self.on_commit_sync)
+
         session_store_data = RunModel(id=self._session_id, blob=self.to_json())
-        database.add(session_store_data)
-        database.commit()
+        try:
+            database.add(session_store_data)
+            database.commit()
+        except Exception as e:
+            pass
+        finally:
+            database.close()
+
+    def upload(self):
+        """Upload the session store db to s3"""
+        protocol, _ = get_protocol_and_path(self._remote_path)
+        db_name = _get_dbname()
+        fs = fsspec.filesystem(protocol)
+        with open(self.location, "rb") as file:
+            with fs.open(f"{self._remote_path}/{db_name}", "wb") as s3f:
+                s3f.write(file.read())
+
+    def download(self) -> List[str]:
+        """Download all the dbs from an s3 locations"""
+        protocol, _ = get_protocol_and_path(self._remote_path)
+        fs = fsspec.filesystem(protocol)
+        databases = fs.glob(f"{self._remote_path}/*.db")
+        databases_location = []
+        for database in databases:
+            database_name = os.path.basename(database)
+            database_timestamp = fs.info(database)["LastModified"]
+            isUpdated = _is_db_updated(database_name, database_timestamp)
+            if isUpdated:
```

Comment:
```suggestion
            if _is_db_updated(database_name, database_timestamp):
```

Too much javascript 😛